src/silo/database.test.cpp

```diff
@@ -1,136 +1,76 @@
+      );
 
    silo::common::LineageTreeAndIdMap lineage_tree;
-   if (config.getLineageDefinitionsFilename().has_value()) {
+   if (config.initialization_files.getLineageDefinitionsFilename().has_value()) {
       lineage_tree = silo::common::LineageTreeAndIdMap::fromLineageDefinitionFilePath(
-         config.getLineageDefinitionsFilename().value()
+         config.initialization_files.getLineageDefinitionsFilename().value()
       );
    }
 
-   silo::preprocessing::Preprocessor preprocessor(
-      config, database_config, reference_genomes, std::move(lineage_tree)
+   auto database = std::make_shared<silo::Database>(
+      silo::Database{silo::initialize::Initializer::createSchemaFromConfigFiles(
+         std::move(database_config), std::move(reference_genomes), std::move(lineage_tree)
+      )}
    );
-   return preprocessor.preprocess();
+   std::ifstream input(input_directory / "input.ndjson");
+   silo::append::appendDataToDatabase(*database, silo::append::NdjsonLineReader{input});
+   return database;
 }
 }  // namespace
 
 TEST(DatabaseTest, shouldBuildDatabaseWithoutErrors) {
    auto database{buildTestDatabase()};
 
-   const auto simple_database_info = database.getDatabaseInfo();
+   const auto simple_database_info = database->getDatabaseInfo();
 
    EXPECT_GT(simple_database_info.total_size, 0);
    EXPECT_EQ(simple_database_info.sequence_count, 5);
-   EXPECT_EQ(simple_database_info.number_of_partitions, 2);
-}
-
-TEST(DatabaseTest, shouldSuccessfullyBuildDatabaseWithoutPartitionBy) {
-   const std::filesystem::path input_directory{"./testBaseData/"};
-
-   auto config = PreprocessingConfig::withDefaults();
-   config.overwriteFrom(
-      silo::config::YamlFile::readFile(input_directory / "test_preprocessing_config.yaml")
-         .verify(PreprocessingConfig::getConfigSpecification())
-   );
-
-   const auto database_config = silo::config::DatabaseConfig::getValidatedConfigFromFile(
-      input_directory / "test_database_config_without_partition_by.yaml"
-   );
-
-   const auto reference_genomes =
-      silo::ReferenceGenomes::readFromFile(config.getReferenceGenomeFilename());
-
-   silo::common::LineageTreeAndIdMap lineage_tree;
-   if (config.getLineageDefinitionsFilename().has_value()) {
-      lineage_tree = silo::common::LineageTreeAndIdMap::fromLineageDefinitionFilePath(
-         config.getLineageDefinitionsFilename().value()
-      );
-   }
-
-   silo::preprocessing::Preprocessor preprocessor(
-      config, database_config, reference_genomes, std::move(lineage_tree)
-   );
-   auto database = preprocessor.preprocess();
-
-   const auto simple_database_info = database.getDatabaseInfo();
-
-   EXPECT_GT(simple_database_info.total_size, 0);
-   EXPECT_EQ(simple_database_info.sequence_count, 100);
    EXPECT_EQ(simple_database_info.number_of_partitions, 1);
 }
 
 // NOLINTNEXTLINE(readability-function-cognitive-complexity)
 TEST(DatabaseTest, shouldReturnCorrectDatabaseInfo) {
```

Comment:
This test is pretty much a duplicate of the one above. The only difference is `EXPECT_GT(simple_database_info.total_size, 0)` vs `EXPECT_EQ(simple_info.total_size, 996)`.